<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>frmMain</class>
 <widget class="QWidget" name="frmMain">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>794</width>
    <height>848</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OpenGl - Qt -Direct Mode Drawing</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="1" column="1">
    <widget class="GLWidget" name="widget" native="true">
     <zorder>frame</zorder>
     <zorder>frame</zorder>
     <zorder>frame</zorder>
    </widget>
   </item>
   <item row="0" column="0" rowspan="2">
    <widget class="QFrame" name="frame">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>180</width>
       <height>0</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <widget class="QWidget" name="verticalLayoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>161</width>
        <height>121</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="sizeConstraint">
        <enum>QLayout::SetNoConstraint</enum>
       </property>
       <item>
        <widget class="QPushButton" name="newShapeButton">
         <property name="text">
          <string>New Shape</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="aboutPushButton">
         <property name="text">
          <string>About</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="quitPushButton">
         <property name="text">
          <string>Quit</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QSpinBox" name="sidesBox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>140</y>
        <width>71</width>
        <height>31</height>
       </rect>
      </property>
      <property name="minimum">
       <number>3</number>
      </property>
      <property name="maximum">
       <number>1000</number>
      </property>
      <property name="value">
       <number>5</number>
      </property>
     </widget>
     <widget class="QPushButton" name="sidesButton">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>140</y>
        <width>71</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>Sides</string>
      </property>
     </widget>
     <widget class="QSlider" name="colourB">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>290</y>
        <width>160</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QSlider" name="colourR">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>230</y>
        <width>160</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="value">
       <number>255</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QSlider" name="colourG">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>260</y>
        <width>160</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="colourLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>210</y>
        <width>62</width>
        <height>19</height>
       </rect>
      </property>
      <property name="text">
       <string>Colour</string>
      </property>
     </widget>
     <widget class="QLabel" name="bColourLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>320</y>
        <width>111</width>
        <height>19</height>
       </rect>
      </property>
      <property name="text">
       <string>Border Colour</string>
      </property>
     </widget>
     <widget class="QSlider" name="sColourR">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>340</y>
        <width>160</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="value">
       <number>0</number>
      </property>
      <property name="sliderPosition">
       <number>0</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QSlider" name="sColourG">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>370</y>
        <width>160</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QSlider" name="sColourB">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>400</y>
        <width>160</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QPushButton" name="radiusButton">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>170</y>
        <width>71</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>Radius</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="radiusBox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>170</y>
        <width>71</width>
        <height>31</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>1000</number>
      </property>
      <property name="singleStep">
       <number>10</number>
      </property>
      <property name="value">
       <number>100</number>
      </property>
     </widget>
    </widget>
   </item>
   <item row="1" column="2">
    <widget class="QFrame" name="frame_3">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
    </widget>
   </item>
   <item row="0" column="1" colspan="2">
    <widget class="QFrame" name="frame_2">
     <property name="maximumSize">
      <size>
       <width>574</width>
       <height>150</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <widget class="QDoubleSpinBox" name="scaleSpinBox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>70</width>
        <height>31</height>
       </rect>
      </property>
      <property name="minimum">
       <double>0.000000000000000</double>
      </property>
      <property name="maximum">
       <double>4.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
     <widget class="QSpinBox" name="xTrans">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>71</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QSpinBox" name="yTrans">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>80</y>
        <width>71</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="transButton">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>110</y>
        <width>161</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>Translate x, y</string>
      </property>
     </widget>
     <widget class="QPushButton" name="scaleButton">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>10</y>
        <width>71</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>Scale</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="matXX">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>20</y>
        <width>70</width>
        <height>24</height>
       </rect>
      </property>
      <property name="decimals">
       <number>1</number>
      </property>
      <property name="minimum">
       <double>-99.900000000000006</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="matYX">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>50</y>
        <width>70</width>
        <height>24</height>
       </rect>
      </property>
      <property name="decimals">
       <number>1</number>
      </property>
      <property name="minimum">
       <double>-99.900000000000006</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="matZX">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>80</y>
        <width>70</width>
        <height>24</height>
       </rect>
      </property>
      <property name="decimals">
       <number>1</number>
      </property>
      <property name="minimum">
       <double>-99.900000000000006</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="matXY">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>20</y>
        <width>70</width>
        <height>24</height>
       </rect>
      </property>
      <property name="decimals">
       <number>1</number>
      </property>
      <property name="minimum">
       <double>-99.900000000000006</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="matYY">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>50</y>
        <width>70</width>
        <height>24</height>
       </rect>
      </property>
      <property name="decimals">
       <number>1</number>
      </property>
      <property name="minimum">
       <double>-99.900000000000006</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="matZY">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>80</y>
        <width>70</width>
        <height>24</height>
       </rect>
      </property>
      <property name="decimals">
       <number>1</number>
      </property>
      <property name="minimum">
       <double>-99.900000000000006</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="matXZ">
      <property name="geometry">
       <rect>
        <x>360</x>
        <y>20</y>
        <width>70</width>
        <height>24</height>
       </rect>
      </property>
      <property name="decimals">
       <number>1</number>
      </property>
      <property name="minimum">
       <double>-99.900000000000006</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="matYZ">
      <property name="geometry">
       <rect>
        <x>360</x>
        <y>50</y>
        <width>70</width>
        <height>24</height>
       </rect>
      </property>
      <property name="decimals">
       <number>1</number>
      </property>
      <property name="minimum">
       <double>-99.900000000000006</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="matZZ">
      <property name="geometry">
       <rect>
        <x>360</x>
        <y>80</y>
        <width>70</width>
        <height>24</height>
       </rect>
      </property>
      <property name="decimals">
       <number>1</number>
      </property>
      <property name="minimum">
       <double>-99.900000000000006</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QWidget</extends>
   <header>glwidget.h</header>
   <container>1</container>
   <slots>
    <slot>newSquare()</slot>
    <slot>newCircle()</slot>
    <slot>setSides(int)</slot>
    <slot>translate(double,double)</slot>
    <slot>scale(double)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>aboutPushButton</sender>
   <signal>clicked()</signal>
   <receiver>frmMain</receiver>
   <slot>aboutBut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>94</x>
     <y>117</y>
    </hint>
    <hint type="destinationlabel">
     <x>190</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>quitPushButton</sender>
   <signal>clicked()</signal>
   <receiver>frmMain</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>141</y>
    </hint>
    <hint type="destinationlabel">
     <x>193</x>
     <y>234</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>newShapeButton</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>newCircle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>114</x>
     <y>53</y>
    </hint>
    <hint type="destinationlabel">
     <x>491</x>
     <y>423</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>colourHex()</slot>
  <slot>colourSnake()</slot>
  <slot>showHex(bool)</slot>
  <slot>showSnake(bool)</slot>
  <slot>endit()</slot>
  <slot>widthSlider(int)</slot>
  <slot>panMode(bool)</slot>
  <slot>openImage()</slot>
  <slot>depthSet(int)</slot>
  <slot>clear()</slot>
  <slot>aboutBut()</slot>
  <slot>helpBut()</slot>
  <slot>txt0()</slot>
  <slot>txt0(QString)</slot>
  <slot>txt1()</slot>
  <slot>txt1(QString)</slot>
  <slot>pressmebut()</slot>
  <slot>txt9(QString)</slot>
  <slot>txt99()</slot>
 </slots>
</ui>
